Sources/Date/Date+NYPLAdditions.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+
+/// A date formatter to get date strings formatted per RFC 3339
+/// without incurring in the high cost of creating a new DateFormatter
+/// each time, which would be ~300% more expensive.
+fileprivate let rfc3339DateFormatter: DateFormatter = {
+  let df = DateFormatter()
+  df.locale = Locale(identifier: "en_US_POSIX")
+  df.timeZone = TimeZone(secondsFromGMT: 0)
+  return df
+}()
+
+public extension Date {
+  var rfc3339String: String {
+    rfc3339DateFormatter.dateFormat = "yyyy'-'MM'-'dd'T'HH':'mm':'ss'Z'"
+    return rfc3339DateFormatter.string(from: self)
+  }
+}
+
+@objc public extension NSDate {
+  @objc(dateFromRFC3339String:)
+  class func date(rfc3339String: String) -> NSDate? {
```

Comment:
in swift this feels more natural as an initializer. Does it complicate things a lot more to have that instead of a class method?

Reply:
also, since this is called from ObjC it might be safer to have an optional param. If for some reason a `nil` sneaks in from the ObjC call site, this will crash.

Reply:
Fixed